reactioncommerce/catalyst
# fix(button): add muitheme to button; add Error button; add reactestlib; fix all tests

Signed-off-by: Machiko Yasuda <machiko@reactioncommerce.com>

Resolves #29 - themes
Resolves #16 - docs
Resolves #19 - tests
Impact: **major**  
Type: **bugfix|style|docs|tests**





## Changes

### Component: Button
- add mui theme to Button
- renames danger button to Error button

### Tests
- remove Enzyme, React Test Renderer
- add https://testing-library.com/docs/react-testing-library, and Jest-dom
- create `renderWithProviders` to add mui theme provider to test renderer

### Packages
- adds `react-testing-library`
- adds `jest-dom`
- removes `enzyme`
- removes `react-test-renderer`

### Templates
- Updates test template to use react-testing-lib and jest-dom.

## Breaking changes
- Danger button is now Error button. Change all `color="danger"` to `color="error"`

## Testing/Reviewing
1. Button docs: https://deploy-preview-32--heuristic-perlman-0ef024.netlify.com/#/Base%20Components/Actions/Button
1. Test ConfirmDialog test
1. Test the tests! Make sure the snapshots reflect reality.
1. Make sure I did not mess up the Babel configs.
1. Make sure the component templates test looks okay.
